src/graph/client/graphClient.ts

```diff
@@ -96,6 +102,21 @@ export class GraphClient {
       this.log("disconnect");
     });
 
+    this._socket.on('setPageState', (query: string, errorMsg: string, results: any[], view: string) => {
+      htmlElements.queryInput.value = query;
+      if (!errorMsg) {
+        this.showResults(results);
+
+        if (view === 'json') {
+          this.selectJsonView();
+        } else {
+          this.selectGraphView();
+        }
+      } else {
+        this.setStateError(errorMsg);
```

Comment:
You should still set the view back to what the user had last selected, even if there's an error.